tests/Tests.FeatureManagement/InMemoryFeatureDefinitionProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.FeatureManagement;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Tests.FeatureManagement
+{
+    class InMemoryFeatureDefinitionProvider : IFeatureDefinitionProvider
+    {
+        private IEnumerable<FeatureDefinition> _definitions;
+
+        public InMemoryFeatureDefinitionProvider(IEnumerable<FeatureDefinition> featureDefinitions)
+        {
+            _definitions = featureDefinitions ?? throw new ArgumentNullException(nameof(featureDefinitions));
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public async IAsyncEnumerable<FeatureDefinition> GetAllFeatureDefinitionsAsync()
```

Comment:
We should add a test for this API too.

Reply:
I agree. Having a test for our `ConfigurationFeatureDefinitionProvider.GetAllFeatureDefinitionsAsync` would be handy. I'll add one, but in a different PR to prevent making this one bigger.

Reply:
I went back and looked and we already have a test for that code path: https://github.com/microsoft/FeatureManagement-Dotnet/blob/fddc98b1ef7fd2efbd5d362bc5194c80bab95a8c/tests/Tests.FeatureManagement/FeatureManagement.cs#L430 

It uses our built-in implementation of `IFeatureDefinitionProvider`, but there's no difference between our implementation of `IFeatureDefinitionProvider`, and an external one. The `CustomFeatureDefinitionProvider` test ensures that an external `IFeatureDefinitionProvider` can be wired up. So it seems to me like the two tests together provide the coverage we need.

* `IFeatureDefinitionProvider` methods tested
* custom `IFeatureDefinitionProvider` is registered and consumed successfully